build.sbt

```diff
@@ -849,7 +850,12 @@ lazy val compiler = configureAsSubproject(project)
         --- (base ** "Scaladoc*ModelTest.scala") // exclude test classes that depend on partest
       ).get
     },
-    Compile / scalacOptions += "-Xlint:-deprecation,-inaccessible,-nonlocal-return,-valpattern,-doc-detached,_",
+    Compile / scalacOptions ++= Seq(
+      "-Xlint",
+      "-feature",
+      "-Wconf:cat=deprecation&msg=early initializers:s", // compiler heavily relies upon early initializers
+      "-Wconf:msg=match may not be exhaustive:i",
```

Comment:
@dwijnand I dropped your commit but added this to one of mine. does that work?

Reply:
yep